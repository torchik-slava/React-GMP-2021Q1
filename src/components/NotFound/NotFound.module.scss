.img {
  display: block;
  margin: 0 auto;
  width: 750px;
  height: 488px;
}

.h1 {
  margin-top: 100px;
  font-size: 4rem;
  font-weight: 300;
  color: #fff;
  text-align: center;
}

.navBtn {
  display: flex;
	justify-content: center;
	align-items: center;
  margin: 0 auto;
  width: 265px;
  height: 60px;
  font-size: 2.1rem;
  border: 2px solid #f65261;
	border-radius: 5px;
  color: #f65261;
	text-decoration: none;
	text-transform: uppercase;
}
