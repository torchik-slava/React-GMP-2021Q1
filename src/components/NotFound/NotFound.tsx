import React from "react";
import { Link } from "react-router-dom";
import img404 from "../../assets/images/notFound.jpg";
import styles from "./NotFound.module.scss";

const NotFoundComponent = () => {
  return (
    <>
			<h1 className={styles.h1}>Page Not Found</h1>
      <img src={img404} className={styles.img} alt="not found pic"/>
			<Link to="/" className={styles.navBtn}>Go back to home</Link>
    </>
  );
};

export default NotFoundComponent;
