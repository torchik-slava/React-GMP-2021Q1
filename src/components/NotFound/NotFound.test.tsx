import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import NotFound from "./NotFound";

describe("NotFound", () => {
  it("renders NotFound correctly", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router history={history}>
        <NotFound />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
