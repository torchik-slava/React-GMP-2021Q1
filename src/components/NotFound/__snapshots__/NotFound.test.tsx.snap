// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders NotFound correctly 1`] = `
<DocumentFragment>
  <h1
    class="h1"
  >
    Page Not Found
  </h1>
  <img
    alt="not found pic"
    class="img"
    src="fileMock"
  />
  <a
    class="navBtn"
    href="/"
  >
    Go back to home
  </a>
</DocumentFragment>
`;
